csrc/ir/utils.cpp

```diff
@@ -1363,8 +1363,9 @@ bool mayRequireAllocation(const TensorView* tv, IterDomain* id) {
   // remain size one.
   // - Reduction: Check the original ID, not the promotion, which may
   //   be a reduction ID even though the original ID is not a reduction
-  return !isPartitionedLoop(tv, id) && !isSizeOneDomain(id) &&
-      !id->isReduction() && !id->isStride();
+  return !ir_utils::isMemoryPartitionedAcross(
```

Comment:
@naoyam This essentially gets rid of the check that the ID needs to be in loop domain. I am uncertain why that check was needed. The tests passed with this change.

Another way could be to add device and stream to this condition and use `isPartitionedLoop` otherwise.

Reply:
This looks good. I think originally this was meant to be used with loop domains, so that's why I put that condition. I don't recall anything particular that this should return false for non-loop IDs, but just in case please run `test --diff` as well.